backend/models/followup.ts

```diff
@@ -39,26 +40,18 @@ FollowupSchema.method("renderSimulationResultsEmail", function () {
   return emailRender(EmailCategory.SimulationResults, this)
 })
 
-FollowupSchema.method("sendSimulationResultsEmail", function () {
-  const followup = this
-  return this.renderSimulationResultsEmail()
-    .then((render) => {
-      return sendMail({
-        to: followup.email,
-        subject: render.subject,
-        text: render.text,
-        html: render.html,
-        tags: [EmailCategory.SimulationResults],
-      })
-    })
-    .then((response) => {
-      return followup.postSimulationResultsEmail(response.messageId)
-    })
-    .catch((err) => {
-      console.log("error", err)
-      followup.error = JSON.stringify(err, null, 2)
-      return followup.save()
-    })
+FollowupSchema.method("sendSimulationResultsEmail", async function () {
+  try {
+    const messageId = await renderAndSendEmail(
```

Comment:
En général c'est souvent à éviter d'avoir un `And` dans un nom de fonction. En fait ça break le "Single Responsibility Principle" (et c'est souvent qu'on est pas au bon niveau d'abstraction). 

En effet ici on veut juste envoyer un email en fait, certe pour envoyer un email il faut le render, mais la méthode n'est pas obligé de le savoir. 

Un article qui en parle https://freedium.cfd/https://medium.com/@samanthaming/code-smell-no-and-in-function-name-c83d0fe5d46e

(après on peut trouver des articles sur tout les point de vu hein 😅 )

Reply:
En phase avec ce point de vue, je vais découper la méthode 👍

Reply:
Ah je pensais juste à la renommer hein